portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -46,11 +42,15 @@ public void init() {
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query commentKeysQuery = new Query("Comment").setKeysOnly();
-    PreparedQuery commentKeysResults =  datastore.prepare(commentKeysQuery);
-    for(Entity commentKey : commentKeysResults.asIterable()){
-      datastore.delete(commentKey.getKey());
+    // Verify the deleter is the author of the comment being deleted. If not, early return.
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()
+        || !request
+            .getParameter("authorEmail")
+            .equals(userService.getCurrentUser().getEmail())) {
+      return;
```

Comment:
Here, it would be appropriate to return an unauthorized error (401) to indicate that the user tried to do something they're not permitted to. You can probably set options on the response object to do this.